### PR TITLE
Drops extended

### PR DESCRIPTION
**Pull request by DIProgan**
dropRequires - What extra feat is required to obtain the drop of an enemy such as dipping into lava or taking a spikehit. Many enemies are placed where Samus can't walk but their drops can be reached anyway.

amountOfDrops - What amount of drops an enemy leaves behind

### CODE REVIEW

**Comment by fdtaillefer** [comment]
Is there value to making `dropRequires` different from all the regular `requires` properties? Looks like devs would have to put additional effort to ultimately get less data.

File: region/norfair/east.json [hunk 1]
```diff
@@ -3276,7 +3280,9 @@
         {
           "name": "Yapping Maw",
           "quantity": 3,
-          "betweenNodes": [ 1, 2 ]
+          "betweenNodes": [ 1, 2 ],
+          "dropRequires": "never",
+          "note": "Possible but requires multiple spikehits in most configurations so rarely worth it"
```
Comment (fdtaillefer) [inline region/norfair/east.json]: The drops are obtainable with Grapple, the issue is actually killing them. Not like farming Yapping Maws would ever be viable anyway.

**Comment by DIProgan** [comment]
I'll remake dropRequires into more typical req data as good as I can. Several cases of heatframes going into it.

**Comment by fdtaillefer** [comment]
> 
> 
> I'll remake dropRequires into more typical req data as good as I can. There'll be some heatframes and crap too.

One of the important things missing is all those places where grapple works. I wonder if we could just roll with a rule that forced damage to reach drops is just not supported though. Feels excessively complicated to start juggling that you have to go to this room and trade X health for X Supers or something.

File: region/brinstar/red.json [hunk 2]
```diff
@@ -387,9 +387,14 @@
...
         {
           "name": "Yapping Maw",
-          "quantity": 3,
+          "quantity": 2,
           "betweenNodes": [1,2],
-          "dropRequires": "spikehit"
+          "dropRequires": [ "never" ]
+        },
+        {
+          "name": "Yapping Maw",
```
Comment (fdtaillefer) [inline region/brinstar/red.json]: Nice. Had to think about it a few seconds to figure it out.

**Pull request merged**